inference-engine/tests/functional/plugin/conformance/subgraphs_dumper/src/main.cpp

```diff
@@ -0,0 +1,65 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include <fstream>
+#include <regex>
+
+#include "inference_engine.hpp"
+
+#include "common_test_utils/file_utils.hpp"
+
+#include "ops_cache.hpp"
+#include "gflag_config.hpp"
+
+// TODO: Poor exceptions handling
+int main(int argc, char *argv[]) {
+    uint8_t ret_code = 0;
+
+
+    gflags::ParseCommandLineNonHelpFlags(&argc, &argv, true);
+    if (FLAGS_h) {
+        showUsage();
+        return 0;
+    }
+    auto ie = InferenceEngine::Core();
+
+    auto cache = SubgraphsDumper::OPCache::make_cache();
+
+    if (!CommonTestUtils::directoryExists(FLAGS_input_folders)) {
+        std::string msg = "Input directory (" + FLAGS_input_folders + ") is not exist!";
+        throw std::runtime_error(msg);
+    }
+
+    if (!CommonTestUtils::directoryExists(FLAGS_output_folders)) {
+        std::string msg = "Output directory (" + FLAGS_output_folders + ") is not exist!";
+        throw std::runtime_error(msg);
+    }
+
+    std::vector<std::string> input_folder_content;
+    CommonTestUtils::directoryFileListRecursive(FLAGS_input_folders, input_folder_content);
+
+    auto xml_regex = std::regex(R"(.*\.xml)");
```

Comment:
You can reuse functionality, which I have implemented on my branch yet (
`inline std::vector<std::string> getFileListByPatternRecursive(const std::vector<std::string> &folderPaths,
                                                              const std::regex& pattern) {`)

Reply:
Let's handle it after merge of both PRs,  direct copy pasting from your branch will leads to merge. conflicts. I've created a task 50406 to track it further.